Address Foodcritic rule FC086

americanhanko opened the pull request:
## This PR

- Addresses Foodcritic rule [FC086: Use databag helper methods to load data bag items](http://www.foodcritic.io/#FC086), which was previously skipped.
- Removes some unnecessary parentheses.

## Why

- Violation of this rule caused issues when attempting to implement [chef-vault](https://github.com/chef/chef-vault).
- Parentheses removed for idiomatic purposes.

americanhanko committed:
address FC086

americanhanko committed:
bump vers to 2.1.1

jazaval reviewed (approved):
Looks good, thanks!

ehanlon commented on recipes/bootstrap.rb:
```diff
@@ -1,6 +1,10 @@
 package 'git'
 package 'openssl'
 
+auth_data = data_bag_item node['vsts_agent']['data_bag'], node['vsts_agent']['data_bag_item']
```
Could we make this recipe a little more flexible by allowing a node attribute for Test Kitchen, and then a `chef_vault_item` for production?

```suggestion
auth_data = node['vsts_agent']['pat'] ? node['vsts_agent']['pat'] : chef_vault_item node['vsts_agent']['data_bag'], node['vsts_agent']['data_bag_item']
```

We would need to depend on `chef-vault` in order to get the `chef_vault_item` helper, and update the `kitchen.yml` to interpolate environment variables to override the attribute. 

A benefit of the `chef_vault_item` helper is the fallback mechanism to `data_bag_item`.

Commit:
Add chef-vault support

Co-Authored-By: americanhanko <eric.hanko1@gmail.com>

americanhanko committed:
ChefSpec, README, and chef-vault

- Add better unit tests for launchd and macosx_service resources
- Implement chef-vault-testfixtures gem
- Update README with latest changes and Azure DevOps documentation

americanhanko committed:
uncomment data_bags_path

americanhanko committed:
add json for vault test fixture
